src/tiled/automapper.cpp

```diff
@@ -712,8 +712,18 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListYes = false;
                 bool matchListNo  = false;
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y()))
-                    return false;
+
+                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                    foreach (const TileLayer *comparedTileLayer, listYes) {
+                        if (!comparedTileLayer->contains(x, y))
+                            return false;
```

Comment:
I tried a different fix: simply assuming that tiles outside the set layer are empty, by doing:
```c++
Cell c1;
if (setLayer->contains(x + offset.x(), y + offset.y()))
    c1 = setLayer->cellAt(x + offset.x(), y + offset.y());
```
However, it did not work, and I have no idea why. Can you explain this? I still find all this rather hard to understand. :-/